src/IO/AsciiPCLoader/asciipcloader.cpp

```diff
@@ -0,0 +1,203 @@
+#include <IO/AsciiPCLoader/asciipcloader.hpp>
+
+#include <Core/Asset/FileData.hpp>
+#include <Core/Utils/Attribs.hpp> //logs
+#include <fstream>
+#include <iostream>
+
+using std::ifstream;
+using std::make_unique;
+using std::runtime_error;
+using std::to_string;
+
+const string trajExt( "txt" );
+
+namespace Ra {
+namespace IO {
+
+AsciiPointCloudLoader::AsciiPointCloudLoader() {}
+
+AsciiPointCloudLoader::~AsciiPointCloudLoader() {}
+
+vector<string> AsciiPointCloudLoader::getFileExtensions() const {
+    return vector<string>( { "*." + trajExt } );
+}
+
+bool AsciiPointCloudLoader::handleFileExtension( const string& extension ) const {
+    return extension.compare( trajExt ) == 0;
+}
+
+Ra::Core::Asset::FileData* AsciiPointCloudLoader::loadFile( const string& filename ) {
+    using Ra::Core::Asset::FileData;
+    using Ra::Core::Asset::GeometryData;
+    using Ra::Core::Utils::logERROR;
+    using Ra::Core::Utils::logINFO;
+
+    FileData* fileData = new Ra::Core::Asset::FileData( filename );
+
+    if ( !fileData->isInitialized() ) {
+        delete fileData;
+        LOG( logINFO ) << "Filedata could not be initialized";
+        return nullptr;
+    }
+
+    auto startTime = clock();
+    // a unique name is required by the component messaging system
+    static int id = 0;
+    auto geometry =
+        make_unique<GeometryData>( "TRAJ_PC_" + to_string( ++id ), GeometryData::POINT_CLOUD );
+    if ( geometry == nullptr ) {
+        LOG( logERROR ) << "could not create geometry";
+        return nullptr;
+    }
+
+    geometry->setFrame( Ra::Core::Transform::Identity() );
+
+    ifstream stream( filename );
+    if ( !stream.is_open() ) {
+        throw runtime_error( "Could not open ifstream to path " + filename );
+        return nullptr;
+    }
+    else {
+        LOG( logINFO ) << "---Beginning file loading---";
+    }
+
+    /*reading how data is arranged*/
+    string line;
+    vector<unsigned long> spacepos;
+    unsigned long timepos, xpos, ypos, zpos;
+
+    if ( !getline( stream, line ) ) {
+        delete fileData;
+        LOG( logINFO ) << "file does not contain trajectory data format. aborting";
+        return nullptr;
+    }
+
+    timepos = line.find( "Time" );
+    xpos    = line.find( " X" );
+    ypos    = line.find( " Y" );
+    zpos    = line.find( " Z" );
+
+    if ( timepos == string::npos ) {
```

Comment:
Avoid code duplication.
One option could be to test all positions at once, and throw a generic error message, e.g., `ile does not contain mandatory properties (X, Y, Z, time). aborting`